src/v0/destinations/braze/transform.js

```diff
@@ -256,10 +376,26 @@ function processTrackEvent(messageType, message, destination, mappingJson) {
     partner: BRAZE_PARTNER_NAME,
   };
 
-  let attributePayload = getUserAttributesObject(message, mappingJson);
+  let attributePayload = getUserAttributesObject(message, mappingJson, destination);
   if (attributePayload && Object.keys(attributePayload).length > 0) {
     attributePayload = setExternalIdOrAliasObject(attributePayload, message);
     requestJson.attributes = [attributePayload];
+    if (destination.Config.deduplicationEnabled) {
```

Comment:
looks like duplicate code, can we make it a function?

Reply:
addressed